### ISSUE TITLE
smtp.Address class can't deal with NULL address - breaks bounces

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10370]
|<img alt="iko's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| iko reported|
|-|-|
|Trac ID|trac#541|
|Type|defect|
|Created|2004-03-05 18:51:52Z|

Attachments:

* [twisted-smtp-040305](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/acf/acf10f2c0a9d00345c2f6be783fc7068fe6a6bff/503d8ad9ad4ea7c6f49a4ffe0bc8d46ce31ad502) (918 bytes) - added by iko on 2004-03-05 18:51:52Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__541 541
type__defect defect
reporter__iko iko
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1078512712000000 1078512712000000
changetime__1474264551168303 1474264551168303
version__ 
owner__ 
cc__exarkun cc__iko
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="iko's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|iko commented|
|-|-|

```
#!html
<pre>
The smtp.Address class has changed to always force the existence of a domain
name. This breaks the null address (&lt;>) used to send bounce messages, since it
will get transmogrified into &lt;@your.dom.ain> which is not a valid address.

While there may be reasons to enforce existance of domain name in some places,
this is not the place, since the Address class knows nothing about the use of
the address (To? From? Something else?), and anyway, guessing email domain names
is likely to get you the wrong guess.

This patch changes it back.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
Resolved a while ago
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc commented|
|-|-|

In changeset 9fd621c0d425661bf23332cd1692ed256aa2e0b9
```
#!CommitTicketReference repository="" revision="9fd621c0d425661bf23332cd1692ed256aa2e0b9"
Merge pull request #541 from berdario/py3-imap4-take2

Author: berdario
Reviewer: rodrigc
Fixes: #6289

Port imap4-utf-7 codec implementation to Python 3
```
